src/Extensions/Logging/Internal/Replayable/ReplayableActivityExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Microsoft.Omex.Extensions.Logging.Replayable;
+
+namespace Microsoft.Omex.Extensions.Logging.Internal.Replayable
+{
+	internal static class ReplayableActivityExtensions
+	{
+		private const string ReplayableLogKey = "ReplayableLogKey";
+
+		private static ConcurrentQueue<LogMessageInformation>? GetReplayableLogsInternal(this Activity activity) =>
+			activity.GetCustomProperty(ReplayableLogKey) as ConcurrentQueue<LogMessageInformation>;
+
+		public static IEnumerable<LogMessageInformation> GetReplayableLogs(this Activity activity) =>
+			activity.GetReplayableLogsInternal() ?? Enumerable.Empty<LogMessageInformation>();
+
+		public static void AddReplayLog(this Activity activity, LogMessageInformation logEvent, uint maxReplayedEventsPerActivity)
+		{
+			ConcurrentQueue<LogMessageInformation>? queue = activity.GetReplayableLogsInternal();
```

Comment:
>GetReplayableLogsInternal [](start = 60, length = 25)

Will it make sense to create a method in the activity itself - GetOrAddReplaableLogsInternal() which does the swap (or Lazy) internally, to separate the concerns in the object that is responsible for having the information?

Reply:
Activity class provided with .NET we could not change it